test/model/tools/model_verify_debug_log/main.cpp

```diff
@@ -0,0 +1,162 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <cstdint>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "wiredtiger.h"
+extern "C" {
+#include "wt_internal.h"
+}
+
+#include "model/driver/debug_log_parser.h"
+#include "model/kv_database.h"
+
+/*
+ * Command-line arguments.
+ */
+extern int __wt_optind, __wt_optwt;
+extern char *__wt_optarg;
+
+/*
+ * Configuration.
+ */
+#define ENV_CONFIG "readonly=true,log=(enabled=false)"
+
+/*
+ * usage --
+ *     Print usage help for the program. (Don't exit.)
+ */
+static void
+usage(const char *progname)
+{
+    fprintf(stderr, "usage: %s [OPTIONS]\n\n", progname);
+    fprintf(stderr, "Options:\n");
+    fprintf(stderr, "  -h HOME    specify the database directory\n");
+    fprintf(stderr, "  -j PATH    load the debug log from a JSON file\n");
+    fprintf(stderr, "  -?         show this message\n");
+}
+
+/*
+ * main --
+ *     The main entry point for the test.
+ */
+int
+main(int argc, char *argv[])
+{
+    const char *debug_log_json, *home, *progname;
+    int ch, ret;
+
+    debug_log_json = nullptr;
+    home = nullptr;
+    progname = argv[0];
+
+    /*
+     * Parse the command-line arguments.
+     */
+    __wt_optwt = 1;
+    while ((ch = __wt_getopt(progname, argc, argv, "h:j:?")) != EOF)
+        switch (ch) {
+        case 'h':
+            home = __wt_optarg;
+            break;
+        case 'j':
+            debug_log_json = __wt_optarg;
+            break;
+        case '?':
+            usage(progname);
+            return EXIT_SUCCESS;
```

Comment:
That would be logical, but for whatever reason, our C++ style guidelines currently prohibit unnecessary parentheses with `return`. So unless we'd like to change the guidelines, let's keep it as it is.

Reply:
There is a good reason for this - for certain types, it can result in unnecessary type conversions or copies. (I don't remember the gory details unfortunately, but there was a good reason for this C++-specific change.)

Reply:
That's quite significant then—thank you for letting us know! It's surprising that parentheses like that would cause such difference, but if that's the case, let's keep it...

(The C++ standard requires certain types returned from a function to be passed to the caller without copying, so that, for example, you can create a vector and simply just return it to the caller—I wonder if this is the behavior that would break?)

Reply:
Yeah - the name for that is "guaranteed copy elision", but I don't remember that being the issue. I'll ask someone in Sydney today for more details.

Reply:
OK points go to @jeremythorp and @marcbutler for finding this: https://stackoverflow.com/questions/4762662/are-parentheses-around-the-result-significant-in-a-return-statement. I think rationale should go in the style guide.